libs/vkd3d/resource.c

```diff
@@ -3415,6 +3418,25 @@ bool vkd3d_create_texture_view(struct d3d12_device *device, const struct vkd3d_t
     view_desc.subresourceRange.levelCount = desc->miplevel_count;
     view_desc.subresourceRange.baseArrayLayer = desc->layer_idx;
     view_desc.subresourceRange.layerCount = desc->layer_count;
+
+    /* This is not correct, but it's the best we can do with existing API.
+     * It should at least avoid a scenario where implicit LOD fetch from invalid levels.
+     * TODO: We will need an extension with vkCreateImageView pNext specificing minLODClamp.
+     * It will be trivial to add in RADV at least ... */
+    if (desc->miplevel_clamp >= 1.0f)
+    {
+        uint32_t new_base_level;
+        uint32_t level_offset;
+        uint32_t end_level;
+
+        assert(view_desc.subresourceRange.levelCount != VK_REMAINING_MIP_LEVELS);
```

Comment:
Is there a reason why we have an assert here? This shouldn't be hard at all to handle properly since we could just not change levelCount in that case.

Reply:
Fixed.